src/EditorFeatures/Core.Wpf/Structure/AbstractStructureTaggerProvider.cs

```diff
@@ -126,51 +128,29 @@ protected sealed override ITaggerEventSource CreateEventSource(ITextView textVie
         {
             try
             {
+                var document = documentSnapshotSpan.Document;
+                if (document == null)
+                    return;
+
                 // Let LSP handle producing tags in the cloud scenario
                 if (documentSnapshotSpan.Document.IsInCloudEnvironmentClientContext())
-                {
                     return;
-                }
 
-                var outliningService = TryGetService(context, documentSnapshotSpan);
-                if (outliningService != null)
-                {
-                    var document = documentSnapshotSpan.Document;
-                    var cancellationToken = context.CancellationToken;
-
-                    // Try to call through the synchronous service if possible. Otherwise, fallback
-                    // and make a blocking call against the async service.
-
-                    var blockStructure = outliningService.GetBlockStructure(document, cancellationToken);
+                var outliningService = BlockStructureService.GetService(document);
+                if (outliningService == null)
+                    return;
 
-                    ProcessSpans(
-                        context, documentSnapshotSpan.SnapshotSpan, outliningService,
-                        blockStructure.Spans);
-                }
+                var blockStructure = outliningService.GetBlockStructure(document, context.CancellationToken);
+                ProcessSpans(
+                    context, documentSnapshotSpan.SnapshotSpan, outliningService,
+                    blockStructure.Spans);
             }
             catch (Exception e) when (FatalError.ReportAndPropagateUnlessCanceled(e))
             {
                 throw ExceptionUtilities.Unreachable;
             }
         }
 
-        private static BlockStructureService TryGetService(
-            TaggerContext<TRegionTag> context,
-            DocumentSnapshotSpan documentSnapshotSpan)
-        {
-            var cancellationToken = context.CancellationToken;
-            using (Logger.LogBlock(FunctionId.Tagger_Outlining_TagProducer_ProduceTags, cancellationToken))
```

Comment:
Should we move this telemetry block/event somewhere?

Reply:
IMO, it was useless before.  so moving it elsewhere won't help.  i don't like us recording telemetry that we're obviously not even examining or otherwise paying attention to.  if we record telemetry, it should be something we actually care about, and that means we're actualy trcking it, have goals for it, and pay attention to it as it changes.